src/snakemake/notebook.py

```
@@ -240,10 +240,49 @@ def get_interpreter_exec(self):
+                "def _():",
+                f"    {re.sub(r'(?:#)\n|(?:\n)#', '\n    ', preamble)}",
+                "    return",
+            ]
+        ).replace("\\", r"\\")
+        source_with_preamble = re.sub(
+            '(if __name__ == "__main__":)', rf"{preamble_cell}\n\n\g<1>", self.source
+        )
+
+        fd.write(source_with_preamble.encode())
+
+    def execute_script(self, fname, edit=None):
```

Ruff (0.14.11)

Unused method argument: `edit`

(ARG002)